src/transformers/models/musicgen/modeling_musicgen.py

```diff
@@ -0,0 +1,2055 @@
+                provided, no constraint is applied. This function takes 2 arguments: the batch ID `batch_id` and
+                `input_ids`. It has to return a list with the allowed tokens for the next generation step conditioned
+                on the batch ID `batch_id` and the previously generated tokens `inputs_ids`. This argument is useful
+                for constrained generation conditioned on the prefix, as described in [Autoregressive Entity
+                Retrieval](https://arxiv.org/abs/2010.00904).
+            synced_gpus (`bool`, *optional*, defaults to `False`):
+                Whether to continue running the while loop until max_length (needed for ZeRO stage 3)
+            kwargs:
+                Ad hoc parametrization of `generate_config` and/or additional model-specific kwargs that will be
+                forwarded to the `forward` function of the model. If the model is an encoder-decoder model, encoder
+                specific kwargs should not be prefixed and decoder specific kwargs should be prefixed with *decoder_*.
+
+        Return:
+            [`~utils.ModelOutput`] or `torch.LongTensor`: A [`~utils.ModelOutput`] (if `return_dict_in_generate=True`
+            or when `config.return_dict_in_generate=True`) or a `torch.FloatTensor`.
+
+                If the model is *not* an encoder-decoder model (`model.config.is_encoder_decoder=False`), the possible
+                [`~utils.ModelOutput`] types are:
+
+                    - [`~generation.GreedySearchDecoderOnlyOutput`],
+                    - [`~generation.SampleDecoderOnlyOutput`],
+                    - [`~generation.BeamSearchDecoderOnlyOutput`],
+                    - [`~generation.BeamSampleDecoderOnlyOutput`]
+
+                If the model is an encoder-decoder model (`model.config.is_encoder_decoder=True`), the possible
+                [`~utils.ModelOutput`] types are:
+
+                    - [`~generation.GreedySearchEncoderDecoderOutput`],
+                    - [`~generation.SampleEncoderDecoderOutput`],
+                    - [`~generation.BeamSearchEncoderDecoderOutput`],
+                    - [`~generation.BeamSampleEncoderDecoderOutput`]
+        """
+        if generation_config is None:
+            generation_config = self.generation_config
+
+        decoder_input_ids = kwargs.get("decoder_input_ids")
+        if decoder_input_ids is not None:
+            seq_len = decoder_input_ids.shape[-1] + 1
+        else:
+            seq_len = 1
+
+        if kwargs.get("max_new_tokens") is not None:
+            generation_config.max_new_tokens = kwargs.get("max_new_tokens")
+            generation_config.max_length = kwargs.get("max_new_tokens") + seq_len
+        if kwargs.get("max_length") is not None:
+            generation_config.max_new_tokens = kwargs.get("max_length") - seq_len
+            generation_config.max_length = kwargs.get("max_length")
```

Comment:
I am not entirely sure why we have this, since by default these values should always be in the `generation_config`, and generate updates them [here](https://github.com/hollance/transformers/blob/5c05ae90ea1a01afe11d78c1e143181ee8f7eddd/src/transformers/generation/utils.py#L1349-L1368).

Reply:
(this creates the input ids using `_prepare_decoder_input_ids_for_generation`

Reply:
It updates `generation_config` but not `self.generation_config` (i.e. the attribute is unchanged)